crates/tap-agent/src/agent/sender_account.rs

```diff
@@ -1143,6 +1149,22 @@ pub mod tests {
         (mock_ecrow_subgraph_server, _mock_ecrow_subgraph)
     }
 
+    async fn mock_escrow_subgraph_empty_response() -> (MockServer, MockGuard) {
+        let mock_ecrow_subgraph_server: MockServer = MockServer::start().await;
+        let _mock_ecrow_subgraph = mock_ecrow_subgraph_server
+            .register_as_scoped(
+                Mock::given(method("POST"))
+                    .and(body_string_contains("TapTransactions"))
+                    .respond_with(ResponseTemplate::new(200).set_body_json(json!({ "data": {
+                            "transactions": [],
+                        }
+                    }))),
+            )
+            .await;
+        (mock_ecrow_subgraph_server, _mock_ecrow_subgraph)
+    }
+
```

Comment:
Did you cinsider implementing this in a "lazy" way since it could be called by multiple tests in parallel and is deterministic?